serpentTools/detfile.py

```diff
@@ -0,0 +1,136 @@
+"""Detector file implementation
+
+Responsible for storing data from detector files
+"""
+
+from collections.abc import Mapping
+import warnings
+
+from .messages import deprecated
+from .detectors import Detector
+from .base import SerpentFile, getStream
+
+
+__all__ = ["DetectorFile"]
+
+
+class DetectorFile(SerpentFile, Mapping):
```

Comment:
Might just want to make this inherit from ``dict`` rather than mapping. The ``__setitem__``, ``update``, and `setdefault` actions should be overloaded to ensure that `DetectorFile` is always full of `Detector` objects